apps/andi/lib/andi_web/input_schemas/ingestion_metatadata_form_schema.ex

```diff
@@ -26,14 +26,39 @@ defmodule AndiWeb.InputSchemas.IngestionMetadataFormSchema do
     :targetDataset
   ]
 
+  def extract_from_ingestion_changeset(%Ecto.Changeset{ data: %Andi.InputSchemas.Ingestion{} } = ingestion_changeset) do
+    ingestion_data = ingestion_changeset
+      |> Changeset.apply_changes()
+
+    extracted_data = %{
+      name: ingestion_data.name,
+      sourceFormat: ingestion_data.sourceFormat,
+      targetDataset: ingestion_data.targetDataset,
+      topLevelSelector: ingestion_data.topLevelSelector
+    }
+
+    mapped_errors = ingestion_changeset.errors
+      |> Enum.map(fn
+        {:name, {msg, opts}} -> {:name, {msg, opts}}
+        {:sourceFormat, {msg, opts}} -> {:sourceFormat, {msg, opts}}
+        {:topLevelSelector, {msg, opts}} -> {:topLevelSelector, {msg, opts}}
+        {:targetDataset, {msg, opts}} -> {:targetDataset, {msg, opts}}
+        other_errors -> nil
+    end)
+    |> Enum.reject(&is_nil/1)
+
+
+    changeset(extracted_data)
+    |> Map.put(:errors, mapped_errors)
+  end
+
   def changeset(changes), do: changeset(%__MODULE__{}, changes)
 
-  def changeset(%__MODULE__{} = current, changes) do
+  def changeset(current, changes) do
     current
-    |> cast(changes, @cast_fields, empty_values: [])
-    |> validate_required(@required_fields, message: "is required")
-    |> validate_top_level_selector()
-    |> target_dataset_exists()
+    |> Changeset.cast(changes, @cast_fields, empty_values: [])
```

Comment:
Removed all validations inside the AndiWeb schema. Andi schemas should be responsible for validations.